src/components/pages/AccountPinModal/AccountPinModal.js

```diff
@@ -29,6 +31,7 @@ const AccountPinModal = () => {
   // all child accounts associated with the main parent login account
   const [accounts, setAccounts] = useState(null);
 
+  // triggered on 'X' button or clicking off the modal
   const handleCancel = () => {
     history.push('/login');
     setShowModal(false);
```

Comment:
Right here it looks like you're pushing to a different page which should cause the component to unmount and then you're updating state on the unmounted component.

Reply:
very true, didn't catch that. thanks